src/Shared/Metrics/InstrumentRecorder.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+
+namespace Microsoft.Extensions.Metrics;
+
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable RS0016 // Add public types and members to the declared API
```

Comment:
Why are we hiding these from docs?

Reply:
These were added before the types were made internal. This is all temporary code that will get replaced with Microsoft.Extensions.Metrics before .NET 8 final ships.